packages/guui/dropdown/index.js

```diff
@@ -0,0 +1,242 @@
+        display: 'block',
+    },
+
+    'li a': {
+        display: 'block',
+        padding: '10px 18px 15px 30px',
+    },
+
+    a: {
+        fontSize: '15px',
+        fontFamily: textSans,
+        color: palette.neutral['1'],
+        lineHeight: 1.2,
+        position: 'relative',
+        transition: 'color 80ms ease-out',
+        padding: '6px 10px',
+        margin: '-1px 0 0 0',
+        textDecoration: 'none',
+        ':hover': {
+            textDecoration: 'underline',
+        },
+        ':focus': {
+            textDecoration: 'underline',
+        },
+
+        ':before': {
+            content: '""',
+            borderTop: '1px solid #ededed',
+            display: 'block',
+            position: 'absolute',
+            top: 0,
+            left: '30px',
+            right: 0,
+        },
+    },
+
+    // no top border for first item
+    'li:first-child a:before': {
+        content: 'none',
+    },
+
+    'a:hover': {
+        backgroundColor: '#ededed',
+        textDecoration: 'none',
+    },
+
+    'a.active': {
+        fontWeight: 'bold',
+
+        ':after': {
+            content: '""',
+            border: '2px solid #c70000',
+            borderTop: 0,
+            borderRight: 0,
+            position: 'absolute',
+            top: '14px',
+            left: '10px',
+            width: '10px',
+            height: '4px',
+            transform: 'rotate(-45deg)',
+        },
+    },
+});
+
+const NoJSCheckbox = styled('input')({
+    ...screenReaderOnly,
+    ':checked + ul': {
+        display: 'block',
+    },
+});
+
+export default class Dropdown extends Component<Props> {
+    constructor(props: Props) {
+        super(props);
+        this.state = { isExpanded: false, noJS: true };
+        this.toggle = this.toggle.bind(this);
+    }
+
+    // hide on ESC key
+    componentDidMount() {
+        const dismiss = event => {
+            const escKey = 27;
+            if (event.keyCode === escKey) {
+                this.setState(() => ({
+                    isExpanded: false,
+                }));
+            }
+        };
+
+        document.addEventListener('keydown', dismiss, false);
+
+        // If componentDidMount runs we know client-side JS is enabled
+        // eslint-disable-next-line react/no-did-mount-set-state
+        this.setState({
+            noJS: false,
+        });
+    }
+
+    toggle() {
+        this.setState(prevState => ({
+            isExpanded: !prevState.isExpanded,
+        }));
+    }
+
+    render() {
+        const { label, links } = this.props;
+
+        // needs to be unique to allow multiple dropdowns on same page
+        // this should be unique because JS is single-threaded
+        const dropdownID = `dropbox-id-${Date.now()}`;
+        const checkboxID = `checkbox-id-${Date.now()}`;
+
+        return (
+            <Div>
+                {this.state.noJS ? (
+                    [
+                        // eslint-disable-next-line jsx-a11y/label-has-for
```

Comment:
I don't understand why this is needed as I have got an htmlFor attribute on the label. So thoughts welcome!?

Also, I think I'm supposed to have a tab index on the label too, but that fails linting rules (this one https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-noninteractive-tabindex.md) so would be great to get thoughts on that too.